user-auth-api.py

```
@@ -0,0 +1,108 @@
+@app.route('/upload', methods=['POST'])
+def upload_file():
+    # Path traversal vulnerability
+    filename = request.form.get('filename')
+    content = request.files['file'].read()
+    
+    # No file validation
+    with open(f"/var/uploads/{filename}", 'wb') as f:
+        f.write(content)
+    
+    # Command injection
+    os.system(f"virus_scan {filename}")
```

Ruff (0.12.2)

Starting a process with a shell, possible injection detected

(S605)